service/service.go

```diff
@@ -189,6 +194,32 @@ func New(config Config) (*Service, error) {
 		}
 	}
 
+	var azureClusterController *controller.AzureCluster
+	{
+		c := controller.AzureClusterConfig{
+			K8sClient: k8sClient,
+			Logger:    config.Logger,
+
+			Flag:  config.Flag,
+			Viper: config.Viper,
```

Comment:
I know there are a lot of new parameters, but I think it's good to reduce the dependency on `viper` by passing the required parameters instead of the whole thing. It allows us to isolate the `viper` dependency, making it easier to change in the future. And making the controller easier to test, because we'd need just the string parameters.